packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx

```diff
@@ -66,6 +68,18 @@ type ChatMessageComponentAsMessageBubbleProps = {
    * @beta
    */
   onDisplayDateTimeString?: (messageDate: Date) => string;
+  /* @conditional-compile-remove(teams-inline-images) */
+  /**
+   * Optional function to fetch attachments.
+   * @beta
+   */
+  onFetchAttachments?: (attachment: FileMetadata) => Promise<void>;
+  /* @conditional-compile-remove(teams-inline-images) */
+  /**
+   * Optional map of attachment ids to blob urls.
+   * @beta
+   */
+  attachmentsMap?: Record<string, string>;
```

Comment:
As an update we will need to be a little smarter with this map. Right now any changes will trigger all message components to re-render. 
Suggestion:  Maybe we create a map of maps, keyed by message id. 
```ts
Record<string, [Record<string, string>]>
```

Reply:
I think it'd require a bigger change that may change the api to also take in the message or messageId for onFetchAttachments. Probably should be a separate task for optimization

Reply:
I agree with this. When we start the e2e implementation we can begin optimizing.